conda_build/metadata.py

```diff
@@ -1358,24 +1388,26 @@ def get_value(self, name, default=None, autotype=True):
         :return: The named value from meta.yaml
         """
         names = name.split("/")
-        assert len(names) in (2, 3), "Bad field name: " + name
         if len(names) == 2:
             section, key = names
             index = None
         elif len(names) == 3:
             section, index, key = names
-            assert section == "source", "Section is not a list: " + section
+            if section not in OPTIONALLY_ITERABLE_FIELDS:
```

Comment:
Unfortunate name, given how dicts are also iterable, technically speaking. Not that we should change it, but I wonder if `MAYBE_LIST_OF_DICTS_FIELDS` (or similar) would have been better.